atlasdb-config/src/main/java/com/palantir/atlasdb/http/FailoverFeignTarget.java

```diff
@@ -150,10 +151,13 @@ private void checkAndHandleFailure(RetryableException ex) {
 
 
     private void pauseForBackoff(RetryableException ex) {
-        double pauseTime = Math.pow(
+        double exponentialPauseTime = Math.pow(
                 GOLDEN_RATIO,
                 numSwitches.get() * failuresBeforeSwitching + failuresSinceLastSwitch.get());
-        pauseForBackoff(ex, Math.round(pauseTime));
+        long cappedPauseTime = Math.min(maxBackoffMillis, Math.round(exponentialPauseTime));
+        long pauseTimeWithJitter = ThreadLocalRandom.current().nextLong(cappedPauseTime);
+
+        pauseForBackoff(ex, pauseTimeWithJitter);
     }
 
     @VisibleForTesting
```

Comment:
At the end I added some randomness, with a base of 250ms and a cap of 750ms. Even though this is not an exponential backoff, we just want to make the client wait for the server to finish the election, and this will suffice to avoid the thundering herd problem.